backend/src/main/java/com/google/sps/AddParticipantHelper.java

```diff
@@ -0,0 +1,212 @@
+      participantDatastore.addParticipant(newParticipant);
+      System.out.println("match not found");
+    }
+
+    // Confirm received form input
+    response.setContentType("text/plain;charset=UTF-8");
+    response.getWriter().println("Received form input details and queried!");
+  }
+
+  /** Retrieve JSON body payload and convert to a JSONObject for parsing purposes */
+  private JSONObject retrieveRequestBody(HttpServletRequest request) throws IOException {
+    StringBuilder requestBuffer = new StringBuilder();
+    try {
+      BufferedReader reader = request.getReader();
+      String currentLine;
+      while ((currentLine = reader.readLine()) != null) {
+        requestBuffer.append(currentLine);
+      }
+    } catch (IOException e) {
+      return null;
+    }
+    return new JSONObject(requestBuffer.toString());
+  }
+
+  /** Get a Participant from form inputs */
+  private Participant getParticipantFromInputs(HttpServletResponse response, JSONObject formDetails)
+      throws IOException {
+    // Get username from email
+    String username = usernameService.getUsername();
+    if (username == null) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Could not retrieve email.");
+      return null;
+    }
+
+    // Get endTimeAvailable and startTimeAvailable in milliseconds
+    long endTimeAvailable = formDetails.getLong(REQUEST_END_TIME_AVAILABLE);
+    long startTimeAvailable = clock.millis();
+
+    // Get desired meeting duration
+    int duration = formDetails.getInt(REQUEST_DURATION);
+    if (duration <= 0) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid duration.");
+      return null;
+    }
+
+    // Get personal preference fields
+    String role = formDetails.getString(REQUEST_ROLE);
+    String productArea = formDetails.getString(REQUEST_PRODUCT_AREA);
+    JSONArray interestsJsonArray = formDetails.getJSONArray(REQUEST_INTERESTS);
+    List<String> interests = getListFromJsonArray(interestsJsonArray);
+    MatchPreference matchPreference =
+        MatchPreference.forStringValue(formDetails.getString(REQUEST_MATCH_PREFERENCE));
+
+    long timestamp = System.currentTimeMillis();
+
+    // Create and return new Participant from input parameters
+    return new Participant(
+        username,
+        startTimeAvailable,
+        endTimeAvailable,
+        duration,
+        role,
+        productArea,
+        interests,
+        matchPreference,
+        /* matchId= */ 0,
+        MatchStatus.UNMATCHED,
+        timestamp);
+  }
+
+  /** Retrieve user email address via Users API and parse for username */
+  public static String getUsername() {
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+    return email != null ? email.split("@")[0] : null;
+  }
+
+  /** Get list of Strings from JSONArray */
+  public static List<String> getListFromJsonArray(JSONArray array) {
```

Comment:
private?

Reply:
Actually, I don't need this method anymore because it's in UsernameService now